other/majority_vote_algorithm.py

```diff
@@ -0,0 +1,33 @@
+"""
+This is Booyer-Moore Majority Vote Algorithm. The problem statement goes like:
+Given an integer array of size n, find all elements that appear more than ⌊ n/k ⌋ times.
+We have to solve in O(n) time and O(1) Space.
+URL : https://en.wikipedia.org/wiki/Boyer%E2%80%93Moore_majority_vote_algorithm
+"""
+from collections import Counter
+
+
+def majority_element(total_votes: list[int], max_candidates: int) -> list[int]:
+    """
+    >>> majority_element([1, 2, 2, 3, 1, 3, 2],3)
+    [2]
```

Comment:
OK... I would rename `total_votes` --> `votes` these are all the individual votes "_streaming_" in.

Let's go back to `partition` -> `votes_needed_to_win` instead of my mistaken `max_candates`.

Let's add a few more test cases (everyone wins, nobody wins) to see where the algorithm succeeds or fails.
```suggestion
    >>> majority_element([1, 2, 2, 3, 1, 3, 2], 3)
    [2]
    >>> majority_element([1, 2, 2, 3, 1, 3, 2], 2)
    ?
    >>> majority_element([1, 2, 2, 3, 1, 3, 2], 4)
    ?
    >>> majority_element([1, 2, 2, 3, 1, 3, 2], 3)
    [2]

>>>
```